ttnn/cpp/ttnn/operations/moreh/moreh_adamw/device/moreh_adamw_device_operation.hpp

```diff
@@ -0,0 +1,154 @@
+    static void validate_on_program_cache_miss(const operation_attributes_t&, const tensor_args_t&);
+
+    // Validate the operation when it reuses a program. Usually will have less checks
+    static void validate_on_program_cache_hit(const operation_attributes_t&, const tensor_args_t&);
+
+    // Compute the output shapes based on the operation attributes and tensor args
+    static shape_return_value_t compute_output_shapes(const operation_attributes_t&, const tensor_args_t&);
+
+    // Create the output tensors based on the operation attributes and tensor args
+    static tensor_return_value_t create_output_tensors(const operation_attributes_t&, const tensor_args_t&);
+
+    // API call to map user arguments to operation attributes and tensor args.
+    // This is the only method that is called by the user
+    // The user will be able to call the operation using `tensor_return_value_t output =
+    // ttnn::prim::adamw(param_in)` after the op is registered Keep in mind that the the overload with
+    // `queue_id` argument will be added automatically for primitive operations So, the user can also call this
+    // operation using `tensor_return_value_t output = ttnn::prim::adamw(queue_id, param_in)`
+    static std::tuple<operation_attributes_t, tensor_args_t> invoke(
+        const Tensor& param_in,
+        const Tensor& grad,
+        const Tensor& exp_avg_in,
+        const Tensor& exp_avg_sq_in,
+
+        float lr,
+        float beta1,
+        float beta2,
+        float eps,
+        float weight_decay,
+        uint32_t step,
+        bool amsgrad,
+
+        const std::optional<const Tensor> max_exp_avg_sq_in,
+        const std::optional<const Tensor> param_out,
+        const std::optional<const Tensor> exp_avg_out,
+        const std::optional<const Tensor> exp_avg_sq_out,
+        const std::optional<const Tensor> max_exp_avg_sq_out,
+        // CHECK if memconfg, compute kernel config require
+        const MemoryConfig& mem_config,
+        std::optional<const DeviceComputeKernelConfig> compute_kernel_config);
+
+    // Optional methods
+
+    // In case the operation need a custom hash function, the following method can be implemented
+    /* static tt::stl::hash::hash_t compute_program_hash(
+        const operation_attributes_t&, const tensor_args_t&);
+    */
+
+    // In case the operation needs a custom create_op_performance_model, this method can be implemented
+    /*
+    static operation::OpPerformanceModel create_op_performance_model(
+        const operation_attributes_t&,
+        const tensor_args_t&,
+        tensor_return_value_t&);
+    */
+};
+
+}  // namespace ttnn::operations::adamw
+
+// Register the operation with the ttnn::register_operation API to make it available to the user as
+// ttnn::prim::adamw
+namespace ttnn::prim {
+constexpr auto moreh_adamw =
+    ttnn::register_operation<"ttnn::prim::moreh_adamw", ttnn::operations::moreh::moreh_adamw::MorehAdamWDeviceOperation>();
```

Comment:
```suggestion
constexpr auto moreh_adamw =
    ttnn::register_operation_with_auto_launch_op<"ttnn::prim::moreh_adamw", ttnn::operations::moreh::moreh_adamw::MorehAdamWDeviceOperation>();
```
There seems to be an issue related to the queue in register_operation at the moment. Please use register_operation_with_auto_launch_op instead